### PR TITLE
Speech impedimemt

### PR DESCRIPTION
**Pull request by BobJoga**
## About The Pull Request

Adds a cheems speech impediment (n as m).

## Why It's Good For The Game

cheems

## Changelog
:cl:
add: Added yet another speech impediment quirk.
/:cl:

### CODE REVIEW

File: modular_skyrat/code/datums/speech_mod/impediments.dm [hunk 1]
```diff
@@ -28,3 +28,9 @@
 	speech_args[SPEECH_MESSAGE] = replacetext(speech_args[SPEECH_MESSAGE], "l", "w")
 	speech_args[SPEECH_MESSAGE] = replacetext(speech_args[SPEECH_MESSAGE], "L", "W")
 
+/datum/speech_mod/impediment_cheems
```
Comment (Weblure) [inline modular_skyrat/code/datums/speech_mod/impediments.dm:31]:
```suggestion
/datum/speech_mod/impediment_nm
```

File: modular_skyrat/code/datums/speech_mod/impediments.dm [hunk 2]
```diff
@@ -28,3 +28,9 @@
...
+	soundtext = "mispronouncing \"n\" as \"m\""
+
+/datum/speech_mod/impediment_cheems/handle_speech(datum/source, list/speech_args)
```
Comment (Weblure) [inline modular_skyrat/code/datums/speech_mod/impediments.dm:34]:
```suggestion
/datum/speech_mod/impediment_nm/handle_speech(datum/source, list/speech_args)
```

File: modular_skyrat/code/datums/traits/neutral.dm [hunk 3]
```diff
@@ -66,6 +66,22 @@
 	if(H)
 		H.disable_speech_mod(/datum/speech_mod/impediment_lw)
 
+/datum/quirk/speech_impediment_cheems
```
Comment (Weblure) [inline modular_skyrat/code/datums/traits/neutral.dm:69]:
```suggestion
/datum/quirk/speech_impediment_nm
```

File: modular_skyrat/code/datums/traits/neutral.dm [hunk 4]
```diff
@@ -66,6 +66,22 @@
...
+	name = "Speech impediment (n as m)"
+	desc = "You mispronounce \"n\" as \"m\"."
+	value = 0
+	medical_record_text = "Patient experiences difficulty in pronouncing certain phonemes."
+
+/datum/quirk/speech_impediment_cheems/add()
```
Comment (Weblure) [inline modular_skyrat/code/datums/traits/neutral.dm:75]:
```suggestion
/datum/quirk/speech_impediment_nm/add()
```

File: modular_skyrat/code/datums/traits/neutral.dm [hunk 5]
```diff
@@ -66,6 +66,22 @@
...
+	var/mob/living/carbon/human/H = quirk_holder
+	if(H)
+		H.enable_speech_mod(/datum/speech_mod/impediment_cheems)
```
Comment (Weblure) [inline modular_skyrat/code/datums/traits/neutral.dm:78]:
```suggestion
		H.enable_speech_mod(/datum/speech_mod/impediment_nm)
```

File: modular_skyrat/code/datums/traits/neutral.dm [hunk 6]
```diff
@@ -66,6 +66,22 @@
...
+
+/datum/quirk/speech_impediment_cheems/remove()
```
Comment (Weblure) [inline modular_skyrat/code/datums/traits/neutral.dm:80]:
```suggestion
/datum/quirk/speech_impediment_nm/remove()
```

File: modular_skyrat/code/datums/traits/neutral.dm [hunk 7]
```diff
@@ -66,6 +66,22 @@
...
+	var/mob/living/carbon/human/H = quirk_holder
+	if(H)
+		H.disable_speech_mod(/datum/speech_mod/impediment_cheems)
```
Comment (Weblure) [inline modular_skyrat/code/datums/traits/neutral.dm:83]:
```suggestion
		H.disable_speech_mod(/datum/speech_mod/impediment_nm)
```

File: modular_skyrat/code/datums/speech_mod/impediments.dm (hunk 1 above)
Comment (BobJoga) [reply modular_skyrat/code/datums/speech_mod/impediments.dm:31]: We have a pretty well-done speech impediment system right now. We really just need to add some way to implement speech quirks in a less... clusterfuck way.

**Comment by jjpark-kb** [comment]
as hilarious as this is, we shouldn't add a lot of speech impediments.

**Pull request closed without merging**